### PR TITLE
[HOLD] Moving firewall rules out from Azure Firewall separate rule set

### PR DESCRIPTION
**Pull request by magrande**
THIS IS ON HOLD, near-concurrent updates of ipGroups is a problem in the RP right now.

### CODE REVIEW

File: networking/hub-region.v2.json [hunk 1]
```diff
@@ -469,6 +471,589 @@
...
+                                                "protocolType": "Https",
+                                                "port": 443
+                                            }
+                                        ],
+                                        "targetUrls": [],
+                                        "terminateTLS": false,
+                                        "sourceAddresses": [],
+                                        "fqdnTags": [],
+                                        "webCategories": [],
+                                        "destinationAddresses": [],
```
Comment (ckittel) [inline networking/hub-region.v2.json:874]: Order doesn't match v1, making this look like a diff. when it shouldn't be.  Also `webCategories` and `destinationAddresses` are not in v1

File: networking/hub-region.v2.json [hunk 2]
```diff
@@ -566,403 +1151,11 @@
...
-                                    "targetFqdns": [
-                                        "azure.archive.ubuntu.com",
-                                        "packages.microsoft.com",
-                                        "archive.ubuntu.com",
-                                        "security.ubuntu.com"
-                                    ]
-                                },
-                                {
-                                    "name": "install-azcli",
-                                    "description": "This is required as the Packer VM needs to install Azure CLI. [Step performed in the referenced jump box building process. Not needed if your jump box building process doesn't do this.]",
```
Comment (ckittel) [inline networking/hub-region.v2.json:759]: In this migration, we are loosing these comments which I've been told are valuable.  Now `description` actually is valid in the REST API, but doesn't look like it's in the ARM schemas yet.  Can you bring them over as comments at least so we don't lose them.

**Review by ckittel — CHANGES_REQUESTED** [review]
LMK if you have any questions.  Nice work on the change over

File: networking/hub-region.v2.json (hunk 2 above)
Comment (magrande) [reply networking/hub-region.v2.json:759]: Done !

File: networking/hub-region.v2.json (hunk 1 above)
Comment (magrande) [reply networking/hub-region.v2.json:887]: Fixed

File: networking/hub-region.v2.json [hunk 3]
```diff
@@ -469,6 +471,589 @@
...
+                                                "protocolType": "Https",
+                                                "port": 443
+                                            }
+                                        ],
+                                        "targetUrls": [],
+                                        "terminateTLS": false,
+                                        "sourceAddresses": [],
+                                        "fqdnTags": [],
+                                        "webCategories": [],
+                                        "destinationAddresses": [],
```
[the remark this answers is not in the document]
Comment (magrande) [reply networking/hub-region.v2.json:708]: Fixed

File: networking/hub-region.v1.json [hunk 4]
```diff
@@ -455,6 +457,342 @@
...
+                                "ruleCollectionType": "FirewallPolicyFilterRuleCollection",
+                                "name": "AKSImageBuilder-Requirements",
+                                "action": {
+                                    "type": "Allow"
+                                },
+                                "priority": 400,
+                                "rules": [
+                                        {
+                                            "ruleType": "ApplicationRule",
+                                            "name": "to-azuremanagement",
```
[the remark this answers is not in the document]
Comment (magrande) [reply networking/hub-region.v1.json]: Fixed in rules and in ruleCollections as well

File: networking/hub-region.v1.json [hunk 5]
```diff
@@ -455,6 +457,342 @@
...
+                                            "sourceAddresses": [],
+                                            "fqdnTags": [],
+                                            "webCategories": [],
+                                            "destinationAddresses": [],
+                                            "targetFqdns": [
+                                                "management.azure.com"
+                                            ]
+                                        },
+                                        {
+                                            "ruleType": "ApplicationRule",
```
[the remark this answers is not in the document]
Comment (magrande) [reply networking/hub-region.v1.json]: Fixed

File: networking/hub-region.v1.json [hunk 6]
```diff
@@ -78,6 +78,8 @@
             "[concat(variables('baseFwPipName'), '-02')]"
         ],
         "hubFwName": "[concat('fw-', parameters('location'))]",
+        "fwPoliciesName": "[concat('fw-policies-', parameters('location'))]",
+        "fwPoliciesBaseName": "fw-policies-base",
```
[the remark this answers is not in the document]
Comment (magrande) [reply networking/hub-region.v1.json:82]: I don't get this comment. Those two variables are always placed between hubFwName and hubVNetName in both files. What am I missing ?

[the remark this answers is not in the document]
Comment (magrande) [reply networking/hub-region.v1.json:82]: I don't get this comment. Those two variables are placed between "hubFwName" and "hubVNetName" in all three files. What am I missing ?

[the remark this answers is not in the document]
Comment (ckittel) [reply networking/hub-region.v1.json:82]: Sure, but not in the same order, leading to a unremarkable diff when you compare the files.  The only diffs that should be seen when you compare the files are the changes required to fit the narrative.

File: networking/hub-region.v1.json [hunk 7]
```diff
@@ -455,6 +457,357 @@
...
+                                        "destinationPorts": [
+                                            "123"
+                                        ]
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                }
+          ]
```
Comment (ckittel) [inline networking/hub-region.v1.json:809]:
```suggestion
            ]
```
Indent is off, making an unremarkable difference between the files, which we want to strongly avoid.

File: networking/hub-region.v2.json (hunk 2 above)
Comment (ckittel) [reply networking/hub-region.v2.json:759]: Thanks! Still missing the migration of the `description` to `//` on network rule `allow-ntp`

File: networking/hub-region.v1.json (hunk 7 above)
Comment (ckittel) [reply networking/hub-region.v1.json]: As mentioned above, this change needs to be in v2 as well.

File: networking/hub-region.v2.json (hunk 1 above)
Comment (ckittel) [reply networking/hub-region.v2.json]: `install-terraform` is still off from v1 to v2

**Review by ckittel — CHANGES_REQUESTED** [review]
All prior feedback items are closed.  Only remaining issues are:

* ipGroup cross-dependency test
* moving ipGroup dependency to the ruleCollection they are consumed by
* Not sure if we can figure out anything about that LA workspace thing... guessing not.
* You'll have to merge `main` in to avoid the merge conflicts

(This list is mostly for myself to remind myself where we left off)

**Comment by ckittel** [comment]
So far out of date, just closing.

**Pull request closed without merging**